circuit/src/rns/secp256k1.rs

```diff
@@ -0,0 +1,222 @@
+use super::*;
+
+#[derive(Debug, Clone, PartialEq, Default)]
+/// We implement two structs one for the Secpk256k1 Base Field as the wrong field and one for the Secp256k1 Scalar Field as the wrong field. The native field is the BN256 scalar field.
+/// The reason for implementing both these structs is that ECDSA verification contains operations in both fields.
+/// Struct for the Secp256k1 Base Field as the wrong field. https://github.com/privacy-scaling-explorations/halo2curves/blob/main/src/secp256k1/fp.rs
+///
+/// Secp25k1 aux points
+/// to_add.x: 0x25968a86095277f8a984c449dc3447d5b2007a27b9eece0db6fb9ae69217bae0
+/// to_add.y: 0xdb08482232d4e3cf46c81d9b71d2247b87c83eb80e46c645758b5bfd51a955d8
+/// to_sub.x: 0xf6530b63da8d89214d6a0cfe38f7294317ebe6f8cd408c9617a123c2b0a7b025
+/// to_sub.y: 0x0903073bbd64df08681cf59bf4689b77e18b198eb3833371e39cf322ff8f1de3
+pub struct Secp256k1BaseField4_68;
+
+impl RnsParams<secp256k1Fp, Fr, 4, 68> for Secp256k1BaseField4_68 {
+	fn native_modulus() -> BigUint {
+		BigUint::from_str(
+			"21888242871839275222246405745257275088548364400416034343698204186575808495617",
+		)
+		.unwrap()
+	}
+
+	fn wrong_modulus() -> BigUint {
+		BigUint::from_str(
+			"115792089237316195423570985008687907853269984665640564039457584007908834671663",
+		)
+		.unwrap()
+	}
+
+	// https://www.wolframalpha.com/input?i=115792089237316195423570985008687907853269984665640564039457584007908834671663+mod+21888242871839275222246405745257275088548364400416034343698204186575808495617
+	// https://www.wolframalpha.com/input?i=6350874878119819312338956282401532410528162663560392320966563075029792193578+in+hex
```

Comment:
We can put wolfram alpha link on top of the file and also format them such that they don't take more than 100 chars